tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -234,55 +282,35 @@ func TestNewMetrics(t *testing.T) {
 }
 
 func TestDenials(t *testing.T) {
-	if err := replaceRouteRule(routeReviewsV3Rule); err != nil {
-		t.Fatalf("Could not create replace reviews routing rule: %v", err)
+	if err := visitProductPage(productPageTimeout, http.StatusOK); err != nil {
+		t.Fatalf("Test app setup failure: %v", err)
 	}
-	// hope for stability
-	time.Sleep(30 * time.Second)
 
-	ratings := fqdn("ratings")
-	if err := createMixerRule(global, ratings, denialRule); err != nil {
-		t.Fatalf("Could not create required mixer rule: %v", err)
+	// deny rule will deny all requests to product page unless
+	// ["x-user"] header is set.
+	glog.Infof("Denials: block productpage if x-user header is missing")
```

Comment:
This fundamentally changes the nature of this test. What is the reasoning here? 

If we strip out everything from the tests, we have no check of any complex behavior (nor need for using something like bookinfo as the basis of a test).

Reply:
This is now a denials integration test. I am going to add some labels processing once https://github.com/istio/mixer/pull/1207 in merged.

I agree that previous behavior was complex, however that complexity was orthogonal to the workings on denials. The above test (with addition of labels) tests Mixer / Pilot / Envoy all together.

Reply:
It may have been orthogonal to the mixer feature. But, imho, it is central to Istio as a whole to have something like this just work.

This test was passing (and recently altered specifically to pass with new proxy logic). I'm not sure what the compelling reason to change it is.

Reply:
A different approach to this discussion:

This PR is changing the fundamental way mixer is configured (affecting metrics reporting, denials enforcement, etc.).  Does it make sense to remove tests of those aspects in the same PR where we are switching the logic?

Either we should do this change first, or we should do in subsequently, imho.  But doing it at the same time raises questions of regression and brittleness.

Reply:
We are not changing all the tests. Which tests are we removing apart from TestDenials?

TestDenials is the only test that is changing behavior, and becoming more focussed. I **need** a test that can test the new code path. There is also https://github.com/istio/mixer/pull/1207  that is needed to use labels which are used in the existing test. We can add those tests back.

Apart from that these changes swap metrics in-place, so the new prometheus metrics adapter generates identical metrics.

What else is brittle?